modules/eks/locals.tf

```
@@ -1,3 +1,16 @@
 locals {
   availability_zone_ids = toset([for subnet in data.aws_subnet.this : subnet.availability_zone_id])
-}
+  # Mapping from user-friendly AMI type names to actual AMI prefixes
+  ami_type_prefix_map = {
+    "alinux2"    = "amazon-eks-node"
```

```suggestion
    "alinux2"    = "amazon-eks-*node"
```

to get also `arm64` amazon linux 2 AMIs

---

I like the idea of getting the ARM AMI too.

Should we add something to the code so you don't get an ARM machine by mistake?

---

It's already in place. The ami filter now include the architecture fetched by instance type. Teorically an instance type has a single architecture

---

@smerlos please, test this approach to fetch also amazon linux 2 arm64 AMI

---

### Summary of Changes and Test Results

I have implemented the requested changes in this PR and conducted validation tests with the following configurations:

#### Configurations and AMIs

1. **ARM Instances (m6g.large):**
   - **AMI alinux2:** `ami-04005026162efa9cc`, *amazon-eks-arm64-node-1.29-v20241024*
   - **AMI alinux2023:** `ami-077b89ce09c9977ea`, *amazon-eks-node-al2023-arm64-standard-1.29-v20241024*

2. **x86_64 Instances (m5.large):**
   - **AMI alinux2:** `ami-079df43d4a41a1e6e`, *amazon-eks-node-1.29-v20241024*
   - **AMI alinux2023:** `ami-079df43d4a41a1e6e`, *amazon-eks-node-al2023-x86_64-standard-1.29-v20241024*

#### Test Results

All tests completed successfully across these instance types and AMI combinations, confirming compatibility and performance expectations on Amazon EKS. No issues were found related to stability or performance.

Let me know if any additional adjustments are required.